### PR TITLE
default cam for asset viewer

### PR DESCRIPTION
**Pull request by TimoBuskeKdab**
took pauls approach moved to asset pipeline editor directly
and added headroom to the numbers.

worked for the scene i've tested so far...

### CODE REVIEW

File: tools/assetpipelineeditor/mainwindow.cpp [hunk 1]
```diff
@@ -274,8 +274,15 @@ void MainWindow::setCamera(const QString &name)
...
 {
-    if (m_camera && m_activeCamera == -1)
-        m_camera->viewAll();
+    if (!m_camera || m_activeCamera != -1)
+        return;
+
+    m_camera->viewAll();
+    QVector3D position = m_camera->position();
+    QVector3D viewCenter = m_camera->viewCenter();
+    float sceneExtent = 2.0f * (viewCenter - position).length();
```
Comment (jjcasmar) [inline tools/assetpipelineeditor/mainwindow.cpp]: const QVector3D position...
const QVector3D viewCenter...
const float sceneExtent...

**Comment by mkrus** [comment]
viewAll is asynchronous, how does this work?

File: tools/assetpipelineeditor/mainwindow.cpp [hunk 2]
```diff
@@ -274,8 +274,15 @@ void MainWindow::setCamera(const QString &name)
...
+    m_camera->setFarPlane(10.0f * sceneExtent); // add some headroom to be able to zoom out
+    m_camera->setNearPlane(0.001f * sceneExtent);
 }
```
Comment (lemirep) [inline tools/assetpipelineeditor/mainwindow.cpp]: m_camera->viewAll() doesn't do anything immediately (it sends commands to Qt3D but the actual response is received later). I think you're just getting lucky with the default values of position and viewCenter and the factors you've applied. What you could do however is connect a lambda or slot to the signals QCamera::positionChanged and/or viewCenterChanged which should be emitted once the viewAll command has been processed to recompute the sceneExtent.

Comment (TimoBuskeKdab) [reply tools/assetpipelineeditor/mainwindow.cpp]: thanks for the info. could be i was lucky. i'll check some more scenes then...

**Comment by TimoBuskeKdab** [comment]
added 3 test scenes to the jira ticket and tested.

**Pull request merged**